CMakeLists.txt

```diff
@@ -24,13 +24,37 @@ target_sources(${APP_TARGET}
         main.cpp
 )
 
+list(APPEND mbed_storage_libs)
+
+if("DATAFLASH" IN_LIST MBED_TARGET_LABELS)
```

Comment:
This is actually good point for the migration path. As the old tools use "components_add", the new tools generate labels from "components_add" and we can include what is needed in examples. Please add  this note to our migration guide story so we do not forget

Reply:
`components_add` is to say that the code for that component should be included in the build.
What I added is to selectively include Mbed storage Libraries depending on what component is required by the application.

So `components_add` is still required.